Drivers/SHT30/sht30.c

```diff
@@ -0,0 +1,69 @@
+#include "sht30.h"
+#include <stdio.h>
+#include <stdbool.h>
+
+#define SWITCHBYTES(x)             (((x<<8) | (x>>8)) & 0xFFFF)      //Switches the first and second byte (This is required for all command values)
+
+/* Creates a new sht30 Driver*/
+sht30_t* sht30_init(sht30_t *sht30)
+{
+    reset(sht30);
+    printf("SHT30 is reset.\n");
+    printf("Heating element is enabled: %d\n", isHeaterEnabled(&sht30));
+    return &sht30;
+}
+
+/**
+ *  Allows for private I2C writing with SHT30
+ */
+static void sht30_write(sht30_t *sht30, uint16_t cmd_value, uint8_t num_bytes) 
+{
+    //TODO STM32 I2C write
+}
+
+/* Allows for private I2C reading with SHT30 */
+static bool sht30_read(sht30_t *sht30, uint8_t msg, uint8_t num_bytes)
+{
+    //TODO STM32 I2C read
+}
+
+/* Writes the command value and then reads the status reg */
+static void sht30_handle_status_reg(sht30_t *sht30)
+{
+    sht30_write(sht30, SWITCHBYTES(SHT30_READSTATUS), 2);
+    uint8_t data[3];
+    SHT30read(&data, 3);
+    uint16_t status = data[0];
+    status <<= 8;
+    status |= data[1];
+    sht30->status_reg = status;
+    //IDK how to get a error from this rn lol
```

Comment:
Theres an stm Hal error type that you can use to return errors if you use the hal, just look up stm32 i2c driver on youtube and it should explain it